raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2083,6 +2098,41 @@ private boolean handleApiVersionsResponse(
         );
     }
 
+    private CompletableFuture<RemoveRaftVoterResponseData> handleRemoveVoterRequest(
+        RaftRequest.Inbound requestMetadata,
+        long currentTimeMs
+    ) {
+        RemoveRaftVoterRequestData data = (RemoveRaftVoterRequestData) requestMetadata.data();
+
+        if (!hasValidClusterId(data.clusterId())) {
+            return completedFuture(
+                new RemoveRaftVoterResponseData().setErrorCode(Errors.INCONSISTENT_CLUSTER_ID.code())
```

Comment:
should we include a message saying that the cluster ID is inconsistent (such as having our cluster ID and theirs?)

Reply:
Sure. We don't do that for the other validation (RPC). Let me file an issue to do that for all of the cluster id checks.

Reply:
Updated it for AddVoter and RemoveVoter: https://issues.apache.org/jira/browse/KAFKA-17197

Will fix it for the other RPCs in a different PR.